File: src/main/java/org/activecraft/activecraftcore/manager/AfkManager.java
package org.activecraft.activecraftcore.manager;

import org.activecraft.activecraftcore.ActiveCraftCore;
import org.activecraft.activecraftcore.events.PlayerAfkEvent;
import org.activecraft.activecraftcore.messagesv2.PlayerMessageFormatter;
import org.activecraft.activecraftcore.messagesv2.MessageSupplier;
import org.activecraft.activecraftcore.playermanagement.Profilev2;
import org.activecraft.activecraftcore.utils.ColorUtils;
import org.activecraft.activecraftcore.ActiveCraftCore;
import org.activecraft.activecraftcore.events.PlayerAfkEvent;
import org.activecraft.activecraftcore.messagesv2.MessageSupplier;
import org.activecraft.activecraftcore.messagesv2.PlayerMessageFormatter;
import org.activecraft.activecraftcore.playermanagement.Profilev2;
import org.activecraft.activecraftcore.utils.ColorUtils;
import org.bukkit.Bukkit;
import org.bukkit.ChatColor;
import org.bukkit.entity.Player;

import java.util.function.Predicate;

public class AfkManager {

    public static void setAfk(Player player, boolean afk) {
        Profilev2 profile = Profilev2.of(player);
        MessageSupplier messageSupplier = profile.getMessageSupplier(ActiveCraftCore.getInstance());

        //call event
        PlayerAfkEvent event = new PlayerAfkEvent(profile, afk);
        Bukkit.getPluginManager().callEvent(event);
        if (event.isCancelled()) return;

        //set afk
        profile.setAfk(event.isAfk());
        String tagFormat = messageSupplier.getMessage("command.afk.tag", messageSupplier.getColorScheme().secondary());
        if (event.isAfk()) {
            profile.getDisplayManager().addTags(tagFormat);
        } else {
            profile.getDisplayManager().removeTags(tagFormat);
        }
        String msg = messageSupplier.getFormatted("command.afk." + (event.isAfk() ? "now-afk" : "not-afk"),
                new PlayerMessageFormatter(messageSupplier.getActiveCraftMessage(), profile));
        player.sendMessage(msg);
        if(ActiveCraftCore.getInstance().getMainConfig().isAnnounceAfk())
            Bukkit.getOnlinePlayers().stream()
                    .filter(Predicate.not(player::equals))
                    .forEach(p -> p.sendMessage((ChatColor.GRAY + ColorUtils.removeColorAndFormat(msg))));
    }
}

File: src/main/java/org/activecraft/activecraftcore/manager/AfkManager.kt
package org.activecraft.activecraftcore.manager

import org.activecraft.activecraftcore.ActiveCraftCore
import org.activecraft.activecraftcore.events.PlayerAfkEvent
import org.activecraft.activecraftcore.exceptions.OperationFailureException
import org.activecraft.activecraftcore.messages.MessageSupplier
import org.activecraft.activecraftcore.messages.PlayerMessageFormatter
import org.activecraft.activecraftcore.playermanagement.Profile
import org.activecraft.activecraftcore.utils.removeColorAndFormat
import org.bukkit.Bukkit
import org.bukkit.ChatColor
import org.bukkit.entity.Player

object AfkManager {
    fun setAfk(player: Player, afk: Boolean) {
        val profile = Profile.of(player) ?: throw OperationFailureException()
        val messageSupplier: MessageSupplier = profile.getMessageSupplier(ActiveCraftCore.INSTANCE)!!

        //call event
        val event = PlayerAfkEvent(profile, afk)
        Bukkit.getPluginManager().callEvent(event)
        if (event.cancelled) return

        //set afk
        profile.isAfk = event.afk
        val tagFormat = messageSupplier.getMessage("command.afk.tag", messageSupplier.colorScheme.secondary)
        if (event.afk) {
            profile.displayManager.addTags(tagFormat)
        } else {
            profile.displayManager.removeTags(tagFormat)
        }
        val msg = messageSupplier.getFormatted(
            "command.afk." + if (event.afk) "now-afk" else "not-afk",
            PlayerMessageFormatter(messageSupplier.activeCraftMessage, profile)
        )
        player.sendMessage(msg)
        if (ActiveCraftCore.INSTANCE.mainConfig.announceAfk) Bukkit.getOnlinePlayers()
            .filter { player != it }
            .forEach { it.sendMessage(ChatColor.GRAY.toString() + removeColorAndFormat(msg)) }
    }
}